libraries/transformer/src/main/java/androidx/media3/transformer/DefaultDecoderFactory.java

```diff
@@ -135,6 +135,9 @@ public DefaultCodec createForVideoDecoding(
     if (SDK_INT >= 31 && requestSdrToneMapping) {
       mediaFormat.setInteger(
           MediaFormat.KEY_COLOR_TRANSFER_REQUEST, MediaFormat.COLOR_TRANSFER_SDR_VIDEO);
+    } else if (SDK_INT >= 31 && MimeTypes.VIDEO_DOLBY_VISION.equals(format.sampleMimeType)) {
```

Comment:
Can you share some documentation which explains this code?

Reply:
Dolby Vision is not only a HDR type but also a media type. But it is based on H.264, H.265 or AV1. That is to say, the Dolby Vision stream is base encoded data + Dolby Vision metadata. Dolby Vision decoder will do two things in playback use case: (1) Call base codec, e.g. H.265 decoder, to decode the base encoded data; (2) Apply Dolby Vision metadata to decoded base data (output of step 1) to get Dolby Vision experience.

In editing use case,  Dolby Vision decoder does step 1 only. We don't apply the Dolby Vision metadata because the result of this process is related to the screen characteristic and environment brightness. For editing use case, the Dolby Vision metadata will be generated by Dolby Vision encoder. For transcoding use case, this is a trade off between better quality and consistent result on different devices.

By default, Dolby Vision decoder will do both above step 1 & 2. The purpose of
`mediaFormat.setInteger(MediaFormat.KEY_COLOR_TRANSFER_REQUEST, MediaFormat.COLOR_TRANSFER_HLG);`
is to notify Dolby Vision decoder to do step 1 only.

There is no open specification to explain above logic. But I can share some documentations about Dolby Vision.
1. Dolby Vision official site: https://www.dolby.com/technologies/dolby-vision/
2. Dolby Vision specification and white paper: https://professionalsupport.dolby.com/s/specifications-and-white-papers?language=en_US

Reply:
> For transcoding use case, this is a trade off between better quality and consistent result on different devices.

Which trade off you are talking about?

Reply:
> > For transcoding use case, this is a trade off between better quality and consistent result on different devices.
> 
> Which trade off you are talking about?

For transcoding use case, ideally, we should do above step 1 & 2 for decoding then encoding to another video format. This is "better quality". In practice (in this contribution), we do step 1 only for decoding. The reason is the result of step 2 is related to the screen characteristic and environment brightness. That means user will get different results on different devices. The transcoding result is not "consistent". So "ignoring step 2" is a trade off.

Reply:
I am still not sure of this change. if the decoder givens us dynamic metadata then I think the pipeline will simple ignore it.

Reply:
This comment seems unresolved.

Reply:
This is important for DV editing/transcoding use case. As explained in my first comment,

Dolby Vision is not only a HDR type but also a media type. But it is based on H.264, H.265 or AV1. That is to say, the Dolby Vision stream is base encoded data + Dolby Vision metadata. Dolby Vision decoder will do two things in playback use case: (1) Call base codec, e.g. H.265 decoder, to decode the base encoded data; (2) Apply Dolby Vision metadata to decoded base data (output of step 1) to get Dolby Vision experience.

In editing use case, Dolby Vision decoder does step 1 only. We don't apply the Dolby Vision metadata because the result of this process is related to the screen characteristic and environment brightness. For editing use case, the Dolby Vision metadata will be generated by Dolby Vision encoder. For transcoding use case, this is a trade off between better quality and consistent result on different devices.

By default, Dolby Vision decoder will do both above step 1 & 2. This is correct for playback use case.
But for editing/transcoding use case, Dolby Vision decoder to do step 1 only. (ignoring Dolby Vision metadata.) That is the purpose of
mediaFormat.setInteger(MediaFormat.KEY_COLOR_TRANSFER_REQUEST, MediaFormat.COLOR_TRANSFER_HLG);
which notifies Dolby Vision decoder to do step 1 only. 

Does this answer your question?